### PR TITLE
Preserve allocation domain when inserting segment_set.

### PR DESCRIPTION
**Pull request by wujingyue**
For #2599.

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

File: csrc/preseg_passes/mark_aliases_prepare.cpp [hunk 1]
```diff
@@ -99,6 +100,16 @@ void insertSegmentSetAfter(
...
+  // Inherit the allocation domain from `use_of`. This is important to pass
+  // AliasTest.Bookend_SegmentSetPreservesAllocation.
+  TensorDomain* replayed_domain =
+      TransformReplay::replayCasP(
+          copy, use_of, -1, TransformReplayOptions().replayAllocation())
+          .first;
+  if (replayed_domain->hasAllocation()) {
+    copy->setAllocationDomain(
+        replayed_domain->allocation(), replayed_domain->contiguity());
+  }
```
Comment (jjsjann123) [inline csrc/preseg_passes/mark_aliases_prepare.cpp:112]: so previously when we forgot to update the allocation domain on the intermediate `copy` here, we ended up with two copies?!

**Review by jjsjann123 — APPROVED** [review]
:clap: 
Good catch!

File: csrc/preseg_passes/mark_aliases_prepare.cpp (hunk 1 above)
Comment (wujingyue) [reply csrc/preseg_passes/mark_aliases_prepare.cpp:112]: In some cases, yes. markAliases treats an empty allocation domain as default.

**Pull request merged**